application/models/Survey.php

```diff
@@ -1938,4 +1939,24 @@ public static function replacePolicyLink($dataSecurityNoticeLabel, $surveyId) {
         return $dataSecurityNoticeLabel;
 
     }
+
+    /**
+     * @return bool
+     */
+    public function getHasQuota()
+    {
+        if (!empty($this->quotas)) {
+            return true;
+        }
+        // check if any tokens have Quota completes (quota models deleted)
+        $dynamic = SurveyDynamic::model($this->sid);
+        if ($dynamic->bHaveToken) {
+            $criteria = new CDbCriteria;
+            $criteria->addCondition('tokens.completed = \'Q\'');
```

Comment:
`$criteria->addCondition('tokens.completed = '.Yii::app()->getBd()->quoteValue('Q'));` https://www.yiiframework.com/doc/api/1.1/CDbConnection#quoteValue-detail

Best for DB compat